model/listeners.py

```diff
@@ -69,7 +69,22 @@ def _site_configuration_has_changed(_db, cooldown=1):
         # Update the timestamp.
         now = utc_now()
         earlier = now - datetime.timedelta(seconds=cooldown)
-        sql = "UPDATE timestamps SET finish=(:finish at time zone 'utc') WHERE service=:service AND collection_id IS NULL AND finish<=(:earlier at time zone 'utc');"
+
+        # We use SKIP LOCKED in the query here, to skip this update if any other
+        # process is currently updating the timestamp. This prevents us from waiting
+        # for the other transaction to update the timestamp, only to update it ourselves
+        # it also avoids a possible deadlock where a request calls back into itself
+        # for example during the library registration process.
+        #
```

Comment:
Minor: looks like two sentences have run together here. Suggest rewording:

Using SKIP LOCKED here allows us avoid waiting for another process that is (presumably) already updating the timestamp, only to immediately update it again ourselves after the wait. It also avoids a possible deadlock for cases in which a request results in another call into the app server while that request is still active. For example, during library registration...

Reply:
👌